code/modules/chemistry/Reagents-Misc.dm

```diff
@@ -182,8 +182,12 @@ datum
 				pop(get_turf(O), amount)
 
 			physical_shock(var/force)
-				if (prob(force*5))
-					explode()
+				if (volume <= holder.total_volume/4) //be somewhat stable to shock if prepared like bang snaps
+					if (prob(max(0,force-12)*12)) //safe to run with, but not sprint. 24% chance to pop on your face when thrown
+						explode()
+				else
+					if (prob(force*5))
+						explode()
```

Comment:
Could probably combine these mathematically a bit so it's a sliding scale rather than there being a magical threshold at 25% concentration? I'm fine with as-is for the sake of simplicity, though.

Reply:
mostly afraid it'd be a little hard to document with more thresholds. especially with how walking and running inflicts random physical shock which'd also cause a lot more inconsistency